allennlp/service/predictors/predictor.py

```diff
@@ -0,0 +1,134 @@
+import json
+import os
+from typing import Dict, Any, Optional, List
+
+from allennlp.common import Params, Registrable
+from allennlp.common.params import replace_none
+from allennlp.data.tokenizers import Tokenizer, WordTokenizer
+from allennlp.data.token_indexers import TokenIndexer
+from allennlp.data.dataset_readers import DatasetReader
+from allennlp.data import Vocabulary
+from allennlp.models import Model
+
+import numpy as np
+import torch
+
+JsonDict = Dict[str, Any]  # pylint: disable=invalid-name
+
+
+def sanitize(x: Any) -> Any:  # pylint: disable=invalid-name
+    """
+    Sanitize x so that it can be JSON serialized
+    """
+    if isinstance(x, (str, float, int, bool)):
+        # x is already serializable
+        return x
+    elif isinstance(x, (torch.Tensor, np.ndarray)):
+        # tensors and arrays need to be converted to lists
```

Comment:
You should probably split the torch tensor and np.ndarrays into two cases, and call `x.cpu().tolist()` on the torch tensor.  Otherwise `x.tolist()` will crash if it was computed on a GPU.